src/rpc/methods/eth/utils.rs

```diff
@@ -97,6 +100,130 @@ where
     }
 }
 
+/// Extract and decode Ethereum revert reason from receipt return data
+pub fn decode_revert_reason(return_data: RawBytes) -> (Vec<u8>, String) {
+    let (data, reason) = match decode_payload(&return_data, CBOR) {
+        Err(e) => {
+            log::warn!("failed to unmarshal cbor bytes from message receipt return error: {e}");
+            (
+                EthBytes::default(),
+                String::from("ERROR: revert reason is not cbor encoded bytes"),
+            )
+        }
+        Ok(data) => (data.clone(), parse_eth_revert(data.0.as_slice())),
+    };
+
+    (data.0, reason)
+}
+
+const ERROR_FUNCTION_SELECTOR: [u8; 4] = [0x08, 0xc3, 0x79, 0xa0]; // Error(string)
+const PANIC_FUNCTION_SELECTOR: [u8; 4] = [0x4e, 0x48, 0x7b, 0x71]; // Panic(uint256)
+
+// Lazily initialized HashMap for panic codes
+static PANIC_ERROR_CODES: Lazy<HashMap<u64, &'static str>> = Lazy::new(|| {
+    let mut m = HashMap::new();
+    m.insert(0x00, "Panic()");
+    m.insert(0x01, "Assert()");
+    m.insert(0x11, "ArithmeticOverflow()");
+    m.insert(0x12, "DivideByZero()");
+    m.insert(0x21, "InvalidEnumVariant()");
+    m.insert(0x22, "InvalidStorageArray()");
+    m.insert(0x31, "PopEmptyArray()");
+    m.insert(0x32, "ArrayIndexOutOfBounds()");
+    m.insert(0x41, "OutOfMemory()");
+    m.insert(0x51, "CalledUninitializedFunction()");
+    m
+});
+
+/// Parse an ABI encoded revert reason from a raw return value.
+///
+/// Handles both `Error(string)` and `Panic(uint256)` formats according to
+/// Solidity's revert conventions.
+///
+/// See https://docs.soliditylang.org/en/latest/control-structures.html#panic-via-assert-and-error-via-require
+fn parse_eth_revert(data: &[u8]) -> String {
+    // If it's not long enough to contain an ABI encoded response, return immediately.
+    if data.len() < 4 + 32 {
```

Comment:
Would be better to use constants like `EVM_WORD_LENGTH`, `EVM_FUNC_SELECTOR_LENGTH`

Reply:
Done.